src/test/scala/ironoxide/FullIntegrationTest.scala

```diff
@@ -65,13 +65,58 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
       val name = GroupName("a name")
       val groupCreateResult = sdk.groupCreate(GroupCreateOpts(validGroupId, name)).attempt.unsafeRunSync.value
 
-      groupCreateResult.id.id.length shouldBe 36
+      groupCreateResult.id.id shouldBe validGroupId.id
       groupCreateResult.name.get.name shouldBe name.name
       groupCreateResult.isAdmin shouldBe true
       groupCreateResult.isMember shouldBe true
       groupCreateResult.created should not be null
       groupCreateResult.lastUpdated shouldBe groupCreateResult.created
-      groupCreateResult.needsRotation shouldBe Some(false)
+      groupCreateResult.needsRotation.value shouldBe false
+    }
+  }
+
+  "Group Add Members" should {
+    "Fail for nonexistent GroupId" in {
+      val sdk = IronSdkSync[IO](deviceContext)
+      val maybeAddMembersResult = sdk.groupAddMembers(GroupId("kumquat"), Nil).attempt.unsafeRunSync
+      maybeAddMembersResult.isLeft shouldBe true
+    }
+    "Fail for adding an existing member" in {
+      val sdk = IronSdkSync[IO](deviceContext)
+      val addMembersResult = sdk.groupAddMembers(validGroupId, List(primaryTestUserId)).attempt.unsafeRunSync.value
+      addMembersResult.failed.head.error should include("User was already a member")
```

Comment:
nitpick: I slightly prefer asserting about the length of the list ahead of doing `head`, which will throw.

Reply:
I added an assertion about length before each call to `head`.